### ISSUE TITLE
Fresh install of v3.2.16 fails to initialize

### ISSUE THREAD

**Issue by StJohn3D** [codota/tabnine-vscode#451]
**please complete the following information:**

- OS version: Win 10
- Editor version: 1.55.0
- Programming language: ts
- TabNine extension version: 3.2.16 & 3.2.15

**If the issue is related to TabNine failure, please provide log files:**

Type `TabNine::config` in your editor, the logs are located at the bottom of the page

**Editor console logs:**

Help -> Toggle Developer Tools

**Issue Details:**

So I tried installing the plugin for the first time today. I saw the 'tabnine' icon and name appear at the bottom of vs code in the status bar. But when I clicked it to open the tabnine hub, the page was simply blank. I tried restarting vs code, uninstalling and re-installing. No code completion was happening, and typing `TabNine::config` did nothing. I then tried rolling back one version to 3.2.15 to see if there was any difference, but the behavior was the same.

I then went all the way back and installed v3.1.26. This time it worked flawlessly. Tabnine hub initialized and I was finally getting completions. I then attempted to upgrade to the latest version, and it is working now.

Based on this, I imagine there is some sort of setup process that the newer version is wrongly expecting to already be there. The newer version probably works for most people if they already have a 3.1.x version installed. However a fresh installation appears to be broken, at least in the two most recent versions as of writing this issue.

**Comment by boaz-codota** [comment]
Wow, thanks for the valuable feedback @StJohn3D !

We will investigate it on our side, but I have some more follow up questions:
1. Are you behind a proxy or do you have some firewall that might block Tabnine's communication?
2. Were you seeing the `Initializing Tabnine` notification?
3. Are you using some ssh development environments or WSL?

**Comment by StJohn3D** [comment]
@boaz-codota no proxy or firewall. I did not see 'Initializing Tabnine' until I went back and installed the 3.1.26 version. The first time I installed the latest version, it just said `tabnine 👈 ` with a hand emoji in the status bar. I clicked it and was taken to the tabnine hub except the entire page was blank. Also the 👈  emoji went away and never came back (assuming that was on purpose just to help me notice it the first time. Nice touch)
Also no ssh or wsl.

**Comment by dimacodota** [comment]
@StJohn3D - if it is not a problem can you please uninstall the extension, reload all vscode sessions (vscode uninstall cleanup is running on restart), reinstall the latest version(3.2.16) and send us the console log ( Help -> Toggle Developer Tools)?
  
Thanks again for the detailed debug session 🙏

**Comment by tbrixey** [comment]
Chiming in to say I am having the same issue. Same versions as John. Let me know if I can help you all out in any way!

**Comment by Jonovono** [comment]
@dimacodota I am having the same issue. Latest version, VSCode Insiders on M1 Macbook Air.

Attached console logs, let me know if this helps?

I also see a blank page if I go to the Tab Nine hub. I saw the suggestions once, but then it stopped, right after a re-boot it showed but then went away following times.

```
console.ts:137 [Extension Host] (node:15977) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead. (at writeOut (internal/process/warning.js:32:3))
m @ console.ts:137
console.ts:137 [Extension Host] Error handling existing version. Falling back to downloading Error: ENOENT: no such file or directory, scandir '/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/binaries' (at Object.handleExistingVersion [as default] (/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/out/binary/binaryFetcher/existingVersionHandler.js:15:17))
m @ console.ts:137
2console.ts:137 [Extension Host] Binary child process exited with code 0 signal unknown (at ChildProcess.<anonymous> (/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/out/binary/Binary.js:76:21))
m @ console.ts:137
console.ts:137 [Extension Host] validating /Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/binaries/3.3.103/aarch64-apple-darwin/TabNine timeout (at Timeout._onTimeout (/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/out/binary/binaryFetcher/binaryValidator.js:14:21))
m @ console.ts:137
console.ts:137 [Extension Host] Binary child process exited with code 0 signal unknown (at ChildProcess.<anonymous> (/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/out/binary/Binary.js:76:21))
m @ console.ts:137
console.ts:137 [Extension Host] validating /Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/binaries/3.3.103/aarch64-apple-darwin/TabNine timeout (at Timeout._onTimeout (/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/out/binary/binaryFetcher/binaryValidator.js:14:21))
m @ console.ts:137
console.ts:137 [Extension Host] validating /Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/binaries/3.3.103/aarch64-apple-darwin/TabNine timeout (at Timeout._onTimeout (/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/out/binary/binaryFetcher/binaryValidator.js:14:21))
m @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:35
_doInvokeHandler @ rpcProtocol.ts:417
_invokeHandler @ rpcProtocol.ts:402
_receiveRequest @ rpcProtocol.ts:318
_receiveOneMessage @ rpcProtocol.ts:245
(anonymous) @ rpcProtocol.ts:110
fire @ event.ts:622
fire @ ipc.net.ts:468
_receiveMessage @ ipc.net.ts:821
(anonymous) @ ipc.net.ts:660
fire @ event.ts:622
acceptChunk @ ipc.net.ts:241
(anonymous) @ ipc.net.ts:202
I @ ipc.net.ts:50
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
2console.ts:137 [Extension Host] Binary child process exited with code 0 signal unknown (at ChildProcess.<anonymous> (/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/out/binary/Binary.js:76:21))
m @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:35
_doInvokeHandler @ rpcProtocol.ts:417
_invokeHandler @ rpcProtocol.ts:402
_receiveRequest @ rpcProtocol.ts:318
_receiveOneMessage @ rpcProtocol.ts:245
(anonymous) @ rpcProtocol.ts:110
fire @ event.ts:622
fire @ ipc.net.ts:468
_receiveMessage @ ipc.net.ts:821
(anonymous) @ ipc.net.ts:660
fire @ event.ts:622
acceptChunk @ ipc.net.ts:241
(anonymous) @ ipc.net.ts:202
I @ ipc.net.ts:50
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
console.ts:137 [Extension Host] validating /Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/binaries/3.3.103/aarch64-apple-darwin/TabNine timeout (at Timeout._onTimeout (/Users/jordan/.vscode-insiders/extensions/tabnine.tabnine-vscode-3.2.16/out/binary/binaryFetcher/binaryValidator.js:14:21))
m @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:35
_doInvokeHandler @ rpcProtocol.ts:417
_invokeHandler @ rpcProtocol.ts:402
_receiveRequest @ rpcProtocol.ts:318
_receiveOneMessage @ rpcProtocol.ts:245
(anonymous) @ rpcProtocol.ts:110
fire @ event.ts:622
fire @ ipc.net.ts:468
_receiveMessage @ ipc.net.ts:821
(anonymous) @ ipc.net.ts:660
fire @ event.ts:622
acceptChunk @ ipc.net.ts:241
(anonymous) @ ipc.net.ts:202
I @ ipc.net.ts:50
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
```

**Comment by Jonovono** [comment]
<img width="1067" alt="Screenshot 2021-04-01 at 1 32 11 pm" src="https://user-images.githubusercontent.com/775808/113332063-b2871100-92ee-11eb-8dcd-e4b9f22e4ddc.png">

Also just got this @dimacodota

**Comment by rossignolli** [comment]
> <img alt="Screenshot 2021-04-01 at 1 32 11 pm" width="1067" src="https://user-images.githubusercontent.com/775808/113332063-b2871100-92ee-11eb-8dcd-e4b9f22e4ddc.png">
> 
> Also just got this @dimacodota

Same problem here. Fresh install Lastest vscode on Windows.

Console log:

``` 
workbench.desktop.main.js:62 [Extension Host] capabilities resolved immediately
workbench.desktop.main.js:62 [Extension Host] validating c:\Users\Vigarani\.vscode\extensions\tabnine.tabnine-vscode-3.2.16\binaries\3.3.103\x86_64-pc-windows-gnu\TabNine.exe timeout
v @ workbench.desktop.main.js:62
$logExtensionHostMessage @ workbench.desktop.main.js:1589
_doInvokeHandler @ workbench.desktop.main.js:1589
_invokeHandler @ workbench.desktop.main.js:1589
_receiveRequest @ workbench.desktop.main.js:1589
_receiveOneMessage @ workbench.desktop.main.js:1589
(anonymous) @ workbench.desktop.main.js:1589
fire @ workbench.desktop.main.js:60
fire @ workbench.desktop.main.js:76
_receiveMessage @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
fire @ workbench.desktop.main.js:60
acceptChunk @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
k @ workbench.desktop.main.js:2793
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:62 [Extension Host] Binary child process stdin error: write EPIPE
v @ workbench.desktop.main.js:62
$logExtensionHostMessage @ workbench.desktop.main.js:1589
_doInvokeHandler @ workbench.desktop.main.js:1589
_invokeHandler @ workbench.desktop.main.js:1589
_receiveRequest @ workbench.desktop.main.js:1589
_receiveOneMessage @ workbench.desktop.main.js:1589
(anonymous) @ workbench.desktop.main.js:1589
fire @ workbench.desktop.main.js:60
fire @ workbench.desktop.main.js:76
_receiveMessage @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
fire @ workbench.desktop.main.js:60
acceptChunk @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
k @ workbench.desktop.main.js:2793
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:62 [Extension Host] Error: Binary request timed out.    at Timeout._onTimeout (c:\Users\Vigarani\.vscode\extensions\tabnine.tabnine-vscode-3.2.16\out\binary\InnerBinary.js:18:24)    at listOnTimeout (internal/timers.js:549:17)    at processTimers (internal/timers.js:492:7)
v @ workbench.desktop.main.js:62
$logExtensionHostMessage @ workbench.desktop.main.js:1589
_doInvokeHandler @ workbench.desktop.main.js:1589
_invokeHandler @ workbench.desktop.main.js:1589
_receiveRequest @ workbench.desktop.main.js:1589
_receiveOneMessage @ workbench.desktop.main.js:1589
(anonymous) @ workbench.desktop.main.js:1589
fire @ workbench.desktop.main.js:60
fire @ workbench.desktop.main.js:76
_receiveMessage @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
fire @ workbench.desktop.main.js:60
acceptChunk @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
k @ workbench.desktop.main.js:2793
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:62 [Extension Host] validating c:\Users\Vigarani\.vscode\extensions\tabnine.tabnine-vscode-3.2.16\binaries\3.3.103\x86_64-pc-windows-gnu\TabNine.exe timeout
v @ workbench.desktop.main.js:62
```

**Comment by dimacodota** [comment]
@rossignolli - can you please send us your tabnine_config.json file to [support@tabnine.com](mailto:support@tabnine.com)?
the file should be located at  `{FOLDERID_RoamingAppData}\TabNine` C:\Users\Alice\AppData\Roaming\TabNine

**Comment by dimacodota** [comment]
@rossignolli @Jonovono @tbrixey  - can you please send us your tabnine_config.json file to support@tabnine.com?
the file should be located at:
window:  `{FOLDERID_RoamingAppData}\TabNine`.   C:\Users\Alice\AppData\Roaming\TabNine
mac :  `$HOME/Library/Preferences/TabNine`    /Users/Alice/Library/Preferences/TabNine
linux: `$XDG_CONFIG_HOME/TabNine` or `$HOME/.config/TabNine`     /home/alice/.config

**Comment by Jonovono** [comment]
@dimacodota just sent, thanks :)

**Comment by rossignolli** [comment]
@dimacodota send it too.

**Comment by dimacodota** [comment]
All, we released a fix (3.3.104 engine version ) please restart and approve

**Comment by rossignolli** [comment]
@dimacodota Everything seems to be working now! Good job!
